scripts/rollup/build.js

```diff
@@ -319,6 +321,15 @@ function getPlugins(
           .replace(/require\(['"]react-is['"]\)/g, "require('ReactIs')");
       },
     },
+    // remove Object.defineProperty(exports, '__esModule', { value: true })
+    // to allow both import React from 'react' and import * as React from 'react'
```

Comment:
This does _not_ allow both unless you're using babelmodules; _please don't write things for babelmodules_ other than maybe in a compatibility bundle. Write things to work on real modules.

Real ES modules will only be able to `import * as React from` from the non-transpiled version, and only be able to `import React from` from the transpiled version. This is why it's important to have a `.mjs` entry point, so that every place is able to do `import * as React from` correctly.

If you want to still preserve `import React from` even in the `.mjs` entry point you're going to need an `export default { ...list of all the exports }`.

Reply:
Reexport with default will be in mjs entry point with NODE_ENV conditions. This is only bundle entry point. This is a hack only  for cjs.